CODINGSTYLE.md

```diff
@@ -0,0 +1,443 @@
+# Coding Style Guide
+
+The Scala.js project has a strict policy regarding coding style.
+This is one of the cornerstones that has allowed Scala.js to maintain a clean, consistent and maintainable codebase.
+
+This document tries to document the style in use as much as possible to make it easier for everyone to contribute.
+
+A *few* of these rules are checked automatically using Scalastyle, but most of them are too complex to teach to an automated tool.
+
+Our coding style is driven and "maintained" by [@sjrd](https://github.com/sjrd), the original author of Scala.js.
+Please follow his directions if they are in conflict with the present document.
+Feel free to point out the conflict, though: we are always looking to improve this document.
```

Comment:
Porting a comment by @yawnt from a commit at https://github.com/sjrd/scala-js/commit/9346a931009c50f5565cd7553903d8ca5bfdaa28#commitcomment-11666243 :

> I think it would be better to start talking about a core team instead of linking everything to a single person. so something like "follow the directions of the core team"
> the rest looks good tho, thumbs up!